src/main/java/ch/njol/skript/expressions/ExprYawPitch.java

```diff
@@ -31,90 +32,133 @@
-		return Number.class;
+	public Class<?>[] acceptChange(final ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE)
+			return CollectionUtils.array(Number.class);
+		return null;
 	}
-	
+
+	@SuppressWarnings("null")
 	@Override
-	protected String getPropertyName() {
-		return yaw ? "yaw" : "pitch";
-	}
-	
-	@SuppressWarnings({"null"})
-		@Override
-		public Class<?>[] acceptChange(final ChangeMode mode) {
-			if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE)
-				return CollectionUtils.array(Number.class);
-			return null;
+	public void change(Event e, Object[] delta, ChangeMode mode) {
+		if (delta[0] == null)
+			return;
+		float value = ((Number) delta[0]).floatValue();
+		Object single = getExpr().getSingle(e);
+		if (single instanceof Location) {
+			changeLocation(((Location) single), value, mode);
+		} else if (single instanceof Vector) {
+			changeVector(e, ((Vector) single), value, mode);
 		}
-	
-		@SuppressWarnings("null")
-		@Override
-		public void change(Event e, Object[] delta, ChangeMode mode) {
-			Location l = getExpr().getSingle(e);
-			if(delta[0] == null || l == null)
-				return;
-			float f = ((Number) delta[0]).floatValue();
-			switch (mode) {
-				case SET:
-					if (yaw)
-						l.setYaw(convertToPositive(f));
-					else
-						l.setPitch(f);
-					break;
-				case ADD:
-					if (yaw)
-						l.setYaw(convertToPositive(l.getYaw()) + f);
-					else
-						l.setPitch(l.getPitch() + f);
-					break;
-				case REMOVE:
-					if (yaw)
-						l.setYaw(convertToPositive(l.getYaw()) - f);
-					else
-						l.setPitch(l.getPitch() - f);
-					break;
+	}
+
+	private void changeLocation(Location l, float value, ChangeMode mode) {
+		switch (mode) {
+			case SET:
+				if (usesYaw)
+					l.setYaw(convertToPositive(value));
+				else
+					l.setPitch(value);
+				break;
+			case ADD:
+				if (usesYaw)
+					l.setYaw(convertToPositive(l.getYaw()) + value);
+				else
+					l.setPitch(l.getPitch() + value);
+				break;
+			case REMOVE:
+				if (usesYaw)
+					l.setYaw(convertToPositive(l.getYaw()) - value);
+				else
+					l.setPitch(l.getPitch() - value);
+				break;
 			default:
 				break;
-			}
 		}
-	
-	
-		//Some random method decided to use for converting to positive values.
-		public float convertToPositive(float f) {
-			if (f != 0 && f * -1 == Math.abs(f))
-				return 360 + f;
-			return f;
-		}	
-	
+	}
+
+	private void changeVector(Event e, Vector vector, float n, ChangeMode mode) {
+		float yaw = VectorMath.getYaw(vector);
+		float pitch = VectorMath.getPitch(vector);
+		switch (mode) {
+			case ADD:
+				if (usesYaw)
+					yaw += n;
+				else
+					pitch -= n; // Negative because of Minecraft's / Skript's upside down pitch
+				vector = VectorMath.fromYawAndPitch(yaw, pitch);
+				getExpr().change(e, new org.bukkit.util.Vector[]{vector}, ChangeMode.SET);
+				break;
+			case REMOVE:
+				n = -n;
```

Comment:
Is this REMOVE supposed to use the reverse SET logic and not the reverse ADD logic?

Reply:
I didnt write this stuff, just copied from the Vector Yaw/Pitch class, so I have no clue to be honest.

Reply:
yeah that seems odd to me... maybe REMOVE should go above ADD?